app/build.gradle

```diff
@@ -1,12 +1,12 @@
 apply plugin: 'com.android.application'
 
 android {
-    compileSdkVersion 24
-    buildToolsVersion "24.0.3"
+    compileSdkVersion 25
+    buildToolsVersion "25.0.0"
     defaultConfig {
         applicationId "org.jak_linux.dns66"
         minSdkVersion 21
-        targetSdkVersion 24
+        targetSdkVersion 25
```

Comment:
I don't see why the changes are necessary. The tile API was introduced in level 24 according to the documentation. And if they really are, they should be in a separate commit prior to the one adding the functionality.